### PR TITLE
FHAC-1080: Add messageID prefix to present message IDs in getAssertion()

### PR DESCRIPTION
**Pull request by danielrf123**
Document Submission Deferred Response does not have prefix included in the BEGIN_OUTBOUND_PROCESSING messageID.  This is happening because the assertion being read in BaseService.getAssertion() already includes the messageID, so no mesageID is "generated", which is where the formatting to include the prefix to the ID is done at.

@TabassumJafri @mhpnguyen please review.

### CODE REVIEW

**Comment by TabassumJafri** [comment]
Endorsing for merge without any comments

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/server/BaseService.java [hunk 1]
```diff
@@ -64,8 +66,13 @@ protected AssertionType getAssertion(WebServiceContext context, AssertionType as
         }
 
         // Extract the message id value from the WS-Addressing Header and place it in the Assertion Class
-        if (assertion != null && NullChecker.isNullish(assertion.getMessageId())) {
-            assertion.setMessageId(extractor.getOrCreateAsyncMessageId(context));
+        if (assertion != null) {
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/server/BaseService.java:69]: [need to confirm]no need to check null b/c the obj is not null when reaching that block.

Comment (danielrf123) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/server/BaseService.java:69]: Could be possible that SAML2AssertionExtractor.getInstance().extractSamlAssertion(context) returns null (line 63).  SonarQube should have complained if its not possible to be null in line 69.

**Comment by mhpnguyen** [comment]
endorse for merging without comments

**Pull request merged**